### PR TITLE
Add testing of Subfiling VFD with zlib compression

### PR DESCRIPTION
**Pull request by hyoklee**
1. Help user to find the directory when test fails.
2. Test zlib compression (OESS-388).

Fixes OESS-388

### CODE REVIEW

File: testpar/t_subfiling_vfd.c [hunk 1]
```diff
@@ -100,18 +100,24 @@ static void test_subfiling_precreate_rank_0(void);
...
+                            test_ioc_only_fail,
+                            test_config_file,
+                            test_stripe_sizes,
+                            test_selection_strategies,
+                            test_read_different_stripe_size,
+                            test_subfiling_precreate_rank_0,
+                            test_subfiling_write_many_read_one,
+                            test_subfiling_write_many_read_few,
+                            test_subfiling_h5fuse
+#ifdef H5_HAVE_FILTER_DEFLATE
```
Comment (byrnHDF) [inline testpar/t_subfiling_vfd.c]: Just add the test to list, and check in the test proper and print "SKIPPED deflate filter not available"

Comment (hyoklee) [reply testpar/t_subfiling_vfd.c]: I want to reduce output message from CMake. I also want to skip building it to speed up compilation.

Comment (byrnHDF) [reply testpar/t_subfiling_vfd.c]: Testing requires consistency in order to compare results across a variety of configurations and platforms. 
I doubt adding the build of a single output statement is going to take that much time.

#ifdef has deflate
 full test
#else
 print SKIPPED - blah blah
#endif

File: testpar/t_subfiling_vfd.c [hunk 2]
```diff
@@ -2116,6 +2121,159 @@ parse_subfiling_env_vars(void)
...
+    curr_nerrors = nerrors;
+
+    dxpl_id = H5Pcreate(H5P_DATASET_XFER);
+    VRFY((dxpl_id >= 0), "DXPL creation succeeded");
+    VRFY((H5Pset_dxpl_mpio(dxpl_id, H5FD_MPIO_COLLECTIVE) >= 0), "H5Pset_dxpl_mpio() succeeded");
+
+    /* Set selection I/O mode on DXPL */
+    VRFY((H5Pset_selection_io(dxpl_id, H5D_SELECTION_IO_MODE_ON) >= 0), "H5Pset_selection_io succeeded");
+
+    if (MAINPROCESS)
```
Comment (byrnHDF) [inline testpar/t_subfiling_vfd.c]: Why isn't this first to be executed?

**Review by byrnHDF — CHANGES_REQUESTED** [review]

File: testpar/t_subfiling_vfd.c (hunk 2 above)
Comment (hyoklee) [reply testpar/t_subfiling_vfd.c]: Good catch - I don't know.
I just copied from the existing function.
@jhendersonHDF , do you know why?

Comment (jhendersonHDF) [reply testpar/t_subfiling_vfd.c]: The above code was added by someone else. Probably best to move the MAINPROCESS prints up everywhere. I can do that after this is merged.

File: testpar/t_subfiling_vfd.c [hunk 3]
```diff
@@ -100,19 +100,19 @@ static void test_subfiling_precreate_rank_0(void);
...
+                            test_ioc_only_fail,
+                            test_config_file,
+                            test_stripe_sizes,
+                            test_selection_strategies,
+                            test_read_different_stripe_size,
+                            test_subfiling_precreate_rank_0,
+                            test_subfiling_write_many_read_one,
+                            test_subfiling_write_many_read_few,
+                            test_subfiling_h5fuse,
+                            t_zlib};
```
Comment (brtnfld) [inline testpar/t_subfiling_vfd.c]: should be test_zlib for consistency

File: testpar/t_subfiling_vfd.c [hunk 4]
```diff
@@ -2116,6 +2115,162 @@ parse_subfiling_env_vars(void)
...
+    buf = NULL;
+
+    VRFY((H5Pclose(plist_id) >= 0), "H5Pclose() succeeded");
+    VRFY((H5Dclose(dset_id) >= 0), "Dataset close succeeded");
+    VRFY((H5Fclose(file_id) >= 0), "File close succeeded");
+
+    mpi_code_g = MPI_Barrier(comm_g);
+    VRFY((mpi_code_g == MPI_SUCCESS), "MPI_Barrier succeeded");
+
+    if (MAINPROCESS) {
```
Comment (brtnfld) [inline testpar/t_subfiling_vfd.c:2221]: Can you also add a test for reading from all the ranks? I think you can just add it to this test.

**Comment by brtnfld** [comment]
Can we get an update on this PR? Where are we at getting it into 1.14.3?

**Comment by brtnfld** [comment]
Adding an extra parameter to each function that turns on/off chunking/compression would be better than copying one of the previous test cases. That way compression is tested for all the cases currently tested for.

File: testpar/t_subfiling_vfd.c (hunk 4 above)
Comment (hyoklee) [reply testpar/t_subfiling_vfd.c:2221]: I added it.

**Comment by hyoklee** [comment]
> Adding an extra parameter to each function that turns on/off chunking/compression would be better than copying one of the previous test cases. That way compression is tested for all the cases currently tested for.

Not for 1.14.3 but for 1.14.4.

**Comment by brtnfld** [comment]
Replaced by #3769

**Pull request closed without merging**